core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -600,6 +600,17 @@ private[spark] class AppStatusListener(
     liveUpdate(task, now)
 
     Option(liveStages.get((event.stageId, event.stageAttemptId))).foreach { stage =>
+      if (event.taskInfo.speculative) {
+        stage.speculationStageSummary = Some(stage.speculationStageSummary
+          .getOrElse(new LiveSpeculationStageSummary(event.stageId, event.stageAttemptId)))
+        val speculationSummary = stage.speculationStageSummary.get
+        speculationSummary.numActiveTasks += 1
+        speculationSummary.numTasks += 1
+      }
+      if(stage.speculationStageSummary.isDefined) {
+        maybeUpdate(stage.speculationStageSummary.get, now)
```

Comment:
We need to do the update only when it is speculative task start - not in other cases.
```suggestion
        maybeUpdate(stage.speculationStageSummary.get, now)
```